src/components/show-products/index.js

```diff
@@ -43,27 +67,46 @@ export const ShowProducts = () => {
 
   return (
     <ContentContainer>
+      <form className="filter-form">
+        <label htmlFor="filter">
+          Filter items
+          <input
+            className="input-item"
+            type="search"
+            id="filter"
+            name="filter"
+            aria-label="Search through list content"
+            placeholder="Start typing here..."
+            value={searchTerm}
+            onChange={(event) => setSearchTerm(event.target.value)}
+          />
+        </label>
+      </form>
       <ul>
-        {products?.map(({ id, productName, lastPurchaseDate, timeFrame }) => (
-          <li className="checkbox-item" key={id}>
-            <label htmlFor={id} className="checkbox-label">
-              <input
-                type="checkbox"
-                id={id}
-                name={productName}
-                onChange={() => handleOnChange(id)}
-                checked={
-                  lastPurchaseDate && compareDates(lastPurchaseDate.toDate())
-                }
-                ariaLabel={timeFrames[timeFrame]}
-              />
-              <span
-                className={`checkmark checkbox-timeFrame-${timeFrame}`}
-              ></span>
-              <span className="checkbox-name">{productName}</span>
-            </label>
-          </li>
-        ))}
+        {list.length > 0 ? (
+          list.map(({ id, productName, lastPurchaseDate, timeFrame }) => (
+            <li className="checkbox-item" key={id}>
+              <label htmlFor={id} className="checkbox-label">
+                <input
+                  type="checkbox"
+                  id={id}
+                  name={productName}
+                  onChange={() => handleCheckboxChange(id)}
+                  checked={
+                    lastPurchaseDate && compareDates(lastPurchaseDate.toDate())
+                  }
+                  aria-label={TimeFrames[timeFrame]}
+                />
+                <span
+                  className={`checkmark checkbox-timeFrame-${timeFrame}`}
+                ></span>
+                <span className="checkbox-name">{productName}</span>
+              </label>
+            </li>
+          ))
+        ) : (
+          <p className="empty-shopping-list">Product not found.</p>
```

Comment:
Only details:

How can you see a message like: `There aren't products that match with 'some search term'`. I believe it could give more feedback about this message to a user,